nalgebra-sparse/src/ops/serial/cs.rs

```diff
@@ -30,30 +32,42 @@ pub fn spmm_cs_prealloc<T>(
 where
     T: Scalar + ClosedAdd + ClosedMul + Zero + One,
 {
+    let some_val = Zero::zero();
+    let mut scratchpad_values: Vec<T> = vec![some_val; b.pattern().minor_dim()];
+    let mut scratchpad_indices: Vec<usize> = vec![0; b.pattern().minor_dim()];
```

Comment:
I agree, this is not really Rust-ic. I was more or less copying from C code (to try to match performance with it), and was somewhat paranoid about Vector implementation in Rust. Didn't want to have any performance artifacts of pushing into the vector. Since then, I read the documentation of the vector and it seems the with_capacity should avoid any variable costs.

Will change this up.